lib/zclient.c

```diff
@@ -1461,6 +1462,223 @@ zebra_interface_vrf_update_read (struct stream *s, vrf_id_t vrf_id,
   *new_vrf_id = new_id;
   return ifp;
 }
+/**
+ * Connect to label manager in a syncronous way
+ *
+ * It first writes the request to zcient output buffer and then
+ * immediately reads the answer from the input buffer.
+ *
+ * @param zclient Zclient used to connect to label manager (zebra)
+ * @result Result of response
+ */
+int
+lm_label_manager_connect (struct zclient *zclient)
```

Comment:
I think you forgot to call this function in ldpd.

BTW I'm not really sure if this ZEBRA_LABEL_MANAGER_CONNECT message is necessary. When a client daemon dies (gracefully or not), the release_daemon_chunks() function is called from zebra_client_close(). Isn't this already enough to prevent leaks of label chunks?